### ISSUE TITLE
iOS: No amount for Credit card & Bancontact but present in Apple Pay with DropIn

### ISSUE THREAD

**Issue by Gp2mv3** [Adyen/adyen-react-native#225]
**Describe the bug**
On iOS, If I give an amount parameter to my AdyenComponent, I don't see any Amount in the payment buttons, only in the "Apple Pay" ActionSheet. The buttons display the text "Add".
If I enter a test card for payment, no amount field is sent to the API; leading to an error.

On Android it works perfectly and displays the text "Pay € XX".

**To Reproduce**
1. Follow the iOS documentation as explained in the README for the DropIn integration. Then, pass this confguration:
```js
{
      amount: {
        value: 2500,
        currency: 'EUR',
      },
      clientKey: XXX,
      shopperLocale: 'FR',
      channel: 'iOS',
      countryCode: 'BE',
      returnUrl: 'cpark://adyencheckout',
      merchantAccount: XXX,
      additionalData: {
        allow3DS2: true,
      },
      dropin: {
        skipListWhenSinglePaymentMethod: true,
        showPreselectedStoredPaymentMethod: false,
      },
      card: {
        holderNameRequired: true,
        showStorePaymentField: false,
      },
      applepay: {
        merchantID: config.adyen.appleMerchantAccount,
        merchantName: 'Seety',
        allowOnboarding: true,
      },
    }
```
2. In your app, show the DropIn menu. You'll have no amount in the payment button for credit cards and Bancontact. If you try Apple Pay, it'll display 25€ as expected.

**Expected behavior**
I expect the amount to be understood the same way as on Android. And the amount field to be displayed, but also sent to the API as needed.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: iPhone 6.5
 - OS: iOS 16.5
 - Version: 1.0.0-rc2

**Comment by descorp** [comment]
Hey @Gp2mv3 

There is known platform misalignment:
iOS SDK operates with `payment` rather with `amount`.

In case you are missing `country code` - right label will not appear on "Pay button".

In JSON above I see `countryCode`, but typo in `currency: 'EUR,` with missing parentally.
Could you doublecheck?

**Comment by Gp2mv3** [comment]
Hello,

Thanks for your answer. Indeed, I miscopied the example to GitHub as I removed a condition there.
It's indeed 'EUR'.

For the misalignment, should we add a payment field then ?

Thanks !

**Comment by descorp** [comment]
> For the misalignment, should we add a payment field then ?

the misalignment is between native iOS and Android libraries. It could be resolved by providing `countryCode` to config.

---

I can't reproduce this with the config you have shared above:

![Screenshot 2023-06-06 at 17 38 25](https://github.com/Adyen/adyen-react-native/assets/2648655/935a6898-e9d4-4508-aab1-a629def7631c)

#worksOnMyMachine

**Comment by Gp2mv3** [comment]
Thanks, I found the error for the button. And it was on my side: I had an old override of "adyen.submitButton" and "adyen.submitButton.formatted" in my localized strings. So now the button displays the right amount, as you show in the screenshot.

However, the onSubmit callback doesn't receive the amount field in paymentData when I confirm the form. Should I add it there before sending the request to the API or is it normal?

**Comment by descorp** [comment]
❗ Never store API Key and never reach out to Adyen API from a client on LIVE. 
All communications should pass through a secure server.

---

> onSubmit callback doesn't receive the amount field

If you see correct "Pay XXX$$$" label - this means the payment is set so the `amount` [should be there](https://github.com/Adyen/adyen-react-native/blob/develop/ios/Helpers/Encodables.swift#L42) 🤔

> Should I add it there before sending the request to the API or is it normal?

We would expect that your backend verify the amount before sending it to Adyen API.
In some case (gift cards and partial payments) it is not necessary, but those are not yet supported by React Native SDK.

**Comment by Gp2mv3** [comment]
Don't worry, we send the requests from our backend. But as it's an account topup, our backend needs to know the amount of payment. So it's needed in the request and is checked on the server side too. 
I have the .amount field in Android but not iOS.

At the moment, I add the amount manually if not present in the onSubmit callback. It works for me but I don't see it in the callback parameters. I'll double check again tomorrow.

**Comment by descorp** [comment]
Sorry @Gp2mv3 

You are indeed right - amount is not populated on iOS in `1.0.0-rc2`.
It was fixed [recently](https://github.com/Adyen/adyen-react-native/commit/637648d)

**Comment by descorp** [comment]
Released on [1.0.0](https://github.com/Adyen/adyen-react-native/releases/tag/1.0.0)

**Comment by descorp** [comment]
I am going to close this thread.
Feel free to continue discussion here and share your feedback.
